profiler/options.go

```diff
@@ -253,3 +276,23 @@ func WithSite(site string) Option {
 		cfg.apiURL = u
 	}
 }
+
+// WithHTTPClient specifies the HTTP client to use when submitting profiles to Site.
+// In general, using this method is only necessary if you have need to customize the
+// transport layer, for instance when using a unix domain socket.
+func WithHTTPClient(client *http.Client) Option {
+	return func(cfg *config) {
+		cfg.httpClient = client
+	}
+}
+
+// WithUDSClient convenience, configures the client to dial a Unix Domain Socket
```

Comment:
Should we provide this or only document how to connect to a UDS using `WithHTTPClient`?

Reply:
This is actually nice but if we want to add it we should add its counterpart in `ddtrace/tracer/option.go`. I'd call it `WithUDS` myself but don't feel strongly on that.

Reply:
added for tracer as well and renamed https://github.com/DataDog/dd-trace-go/pull/788/commits/afcfaab36ac35303cd62a73301d20dd575d6af24